src/middleware/logger.ts

```diff
@@ -0,0 +1,11 @@
+import { type LogArg } from "../helpers/types";
+
+// Based on https://datatracker.ietf.org/doc/html/rfc5424#autoid-11
+// it's pretty reasonable to expect a logger to have the following interfaces
+// available.
+export interface ILogger {
```

Comment:
Nit: the name `ILogger` is Hungarian Notationy. Can we use `Logger` instead?

Reply:
Sure, didn't know that's a thing.